run.py

```diff
@@ -33,17 +37,80 @@ def _get_opt_parser(self):
         parser.add_option('-p', '--pass', dest='password', default=None)
         return parser
 
-    def _get_connection(self):
+    def _get_connection(self, hostname, username, password):
```

Comment:
juju ssh works by reading, on 1.25.x ~/.juju/ssh for keys and in juju2 ~/.local/share/juju/ssh (I believe) so getting the keys locally should be fairly easy!